velox/expression/CastExpr-inl.h

```diff
@@ -285,6 +285,14 @@ void CastExpr::applyCastKernel(
       return;
     }
 
+    if constexpr (
+        (FromKind == TypeKind::BOOLEAN) && ToKind == TypeKind::TIMESTAMP) {
+      const auto castResult =
+          hooks_->castBooleanToTimestamp((bool)inputRowValue);
```

Comment:
Do we need the cast as 'inputRowValue' should be of bool type?

Reply:
I assume the case is different because above `castIntToTimestamp` only supports int64_t and the `inputRowValue` could be int8_t, int16_t, int32_t, int64_t so it needs a cast, while the input for `castBooleanToTimestamp` must be bool.